pkg/http/utils.go

```diff
@@ -397,3 +398,31 @@ func newRetryClient() *retryablehttp.Client {
 	}
 	return retryClient
 }
+
+func GetRequestWithTimeout(url string, timeout time.Duration) ([]byte, error) {
+	// A HTTP client with the specified timeout
+	client := &http.Client{
+		Timeout: timeout,
+	}
+
+	// Create the HTTP request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating module-allowlist request failed: %s", err)
```

Comment:
I think this error is specific to the allowlist feature but the intention is for this client to be generic no? My suggestion here just do `request failed` and the caller will prepend with the appropriate `return fmt.Errorf("failed to fetch module allowlist: %s", err)`

Reply:
Thanks- Totally makes sense!